packages/headless-components/services/src/react/ServiceList.tsx

```diff
@@ -0,0 +1,467 @@
+    const { children } = props;
+    const servicesListService = useService(ServicesListServiceDefinition);
+    const services = servicesListService.services.get();
+    const pagingMetadata = servicesListService.pagingMetadata.get();
+    const displayedServices = services.length;
+    const totalServices = pagingMetadata?.count || services.length;
+    const isFiltered =
+      Object.keys(servicesListService.filters.get()).length > 0;
+
+    return (
+      <div ref={ref}>
+        {typeof children === 'function'
+          ? children({ totalServices, displayedServices, isFiltered })
+          : children}
+      </div>
+    );
+  },
+);
+
+Raw.displayName = 'Services.Raw';
+
+export interface ErrorProps {
+  asChild?: boolean;
+  children?: React.ReactNode;
+  className?: string;
+}
+
+export const Error = React.forwardRef<HTMLDivElement, ErrorProps>(
+  (props: ErrorProps, ref: React.Ref<HTMLDivElement>) => {
+    const { asChild, children, className, ...otherProps } = props;
+
+    return (
+      <CoreServiceListError>
+        {({ error }: { error: string | null }) => {
+          if (!error) {
+            return null;
+          }
+
+          return (
+            <AsChildSlot
+              ref={ref}
+              asChild={asChild}
+              className={className}
+              data-testid={TestIds.serviceError}
+              data-error={error}
+              customElement={children}
+              customElementProps={{
+                error,
+              }}
+              content={error}
+              {...otherProps}
+            >
+              <div className="text-status-error text-sm sm:text-base">
+                {error}
+              </div>
+            </AsChildSlot>
+          );
+        }}
+      </CoreServiceListError>
+    );
+  },
+);
+
+Error.displayName = 'Services.Error';
+
+export const Service = {
+  Root: React.forwardRef<
+    HTMLDivElement,
+    {
+      service: services.Service;
+      children:
+        | ((props: { service: services.Service }) => React.ReactNode)
+        | React.ReactNode;
+      [key: string]: any; // For additional props like data-testid
+    }
+  >((props, ref) => {
+    const { service, children, ...rest } = props;
+
+    return (
+      <ServiceContext.Provider value={{ service }}>
+        <div
+          data-testid={TestIds.serviceRoot}
+          ref={ref as React.Ref<HTMLDivElement>}
+          {...rest}
+        >
+          {typeof children === 'function' ? children({ service }) : children}
+        </div>
+      </ServiceContext.Provider>
+    );
+  }),
+
+  Name: React.forwardRef<HTMLDivElement, { className?: string }>(
+    (props: { className?: string }, ref: React.Ref<HTMLDivElement>) => {
+      const context = React.useContext(ServiceContext);
+      if (!context) {
+        throw new ServiceComponentError(
+          'Service components must be used within a Service.Root component',
+        );
+      }
+      const { service } = context;
```

Comment:
We prefer to create a service for the entity and a "Service" core component consuming that service and exposing the different fields (such as "Name", "Description" and so on). Why did you decide to use context instead?